tools/report-converter/codechecker_report_converter/sparse/output_parser.py

```diff
@@ -0,0 +1,106 @@
+# -------------------------------------------------------------------------
+#
+#  Part of the CodeChecker project, under the Apache License v2.0 with
+#  LLVM Exceptions. See LICENSE for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# -------------------------------------------------------------------------
+
+import logging
+import os
+import re
+
+from ..output_parser import BaseParser, Message, Event
+LOG = logging.getLogger('ReportConverter')
+
+
+class SparseParser(BaseParser):
+    """
+    Parser for Sparse Output
+    """
+
+    def __init__(self, analyzer_result):
+        super(SparseParser, self).__init__()
+
+        self.analyzer_result = analyzer_result
+
+        self.message_line_re = re.compile(
+            # File path followed by a ':'.
+            r'^(?P<path>[\S ]+?):'
+            # Line number followed by a ':'.
+            r'(?P<line>\d+?):'
+            # Column number followed by a ':'.
+            r'(?P<column>\d+?):'
+            # Message.
+            r'(?P<message>[\S \t]+)\s*')
+
+        self.multi_line_message_re = re.compile(
+            r'^(?P<path>[\S ]+?):'
+            r'(?P<message>[\S \t]+)\s*'
+        )
+
+        self.note_line_re = re.compile(
+            r'^(?P<path>\.[\S ]+?):'
+            # Line number followed by a ':'.
+            r'(?P<line>\d+?):'
+            # Column number followed by a ':'.
+            r'(?P<column>\d+?):'
+            # Message.
+            r'(?P<message>[\S \t]+)\s*'
+        )
+
+    def parse_message(self, it, line):
+        """
+        Actual Parsing function for the given line
+        It is expected that each line contains a seperate report
+        """
+        match = self.message_line_re.match(line)
+        multi_line_match = self.multi_line_message_re.match(line)
+
+        if (match is None) and (multi_line_match is None):
+            return None, next(it)
+
+        checker_name = None
+
+        if match:
+            file_path = os.path.normpath(
+                os.path.join(os.path.dirname(self.analyzer_result),
+                             match.group('path')))
+            message = Message(
+                file_path,
+                int(match.group('line')),
+                int(match.group('column')),
+                match.group('message').strip(),
+                checker_name)
+
+        else:
+            file_path = os.path.normpath(
+                os.path.join(os.path.dirname(self.analyzer_result),
+                             multi_line_match.group('path')))
+
+            message = Message(
+                file_path,
+                0,
+                0,
+                multi_line_match.group('message').strip(),
+                checker_name)
```

Comment:
We could simply remove this else statement right??

Reply:
Yes, we can remove it and remove all the code which references the multi_line_match variable.

Reply:
@csordasmarton removing this and the above-mentioned regex was actually creating a separate `sample.h.plist` file and not appending it to `sample.c`. Nevertheless, I have modified the code to not include the `files/sample.c: note: in included file:` line in the output